NuKeeper.Update.Tests/UpdateRunnerTests.cs

```diff
@@ -0,0 +1,218 @@
+using NSubstitute;
+using NuKeeper.Abstractions.Logging;
+using NuKeeper.Update.Process;
+using NUnit.Framework;
+using NuKeeper.Abstractions.NuGet;
+using System.Threading.Tasks;
+
+namespace NuKeeper.Update.Tests
+{
+    [TestFixture]
+    public class UpdateRunnerTests
+    {
+        private INuKeeperLogger _nuKeeperLogger;
+        private ISettingsContainer _settingsContainer;
+        private IFileRestoreCommand _fileRestoreCommand;
+        private INuGetUpdatePackageCommand _nuGetUpdatePackageCommand;
+        private IDotNetUpdatePackageCommand _dotNetUpdatePackageCommand;
+        private IDotNetRestoreCommand _dotNetRestoreCommand;
+        private IUpdateProjectImportsCommand _updateProjectImportsCommand;
+        private IUpdateNuspecCommand _updateNuspecCommand;
+        private IUpdateDirectoryBuildTargetsCommand _updateDirectoryBuildTargetsCommand;
+        private IUpdateRunner _sut;
+
+        [OneTimeSetUp]
+        public void SetUp()
+        {
+            _nuKeeperLogger = Substitute.For<INuKeeperLogger>();
+
+            _settingsContainer = Substitute.For<ISettingsContainer>();
+            _settingsContainer.UserSettings = new UserSettings();
+
+            _fileRestoreCommand = Substitute.For<IFileRestoreCommand>();
+            _nuGetUpdatePackageCommand = Substitute.For<INuGetUpdatePackageCommand>();
+            _dotNetUpdatePackageCommand = Substitute.For<IDotNetUpdatePackageCommand>();
+            _dotNetRestoreCommand = Substitute.For<IDotNetRestoreCommand>();
+            _updateProjectImportsCommand = Substitute.For<IUpdateProjectImportsCommand>();
+            _updateNuspecCommand = Substitute.For<IUpdateNuspecCommand>();
+            _updateDirectoryBuildTargetsCommand = Substitute.For<IUpdateDirectoryBuildTargetsCommand>();
+
+            _sut = new UpdateRunner(
+                _nuKeeperLogger,
+                _settingsContainer,
+                _fileRestoreCommand,
+                _nuGetUpdatePackageCommand,
+                _dotNetUpdatePackageCommand,
+                _dotNetRestoreCommand,
+                _updateProjectImportsCommand,
+                _updateNuspecCommand,
+                _updateDirectoryBuildTargetsCommand);
+        }
+
+        [TestCase(true, Description = "Should perform dotnet restore's before each package update.")]
+        [TestCase(false, Description = "Should not perform dotnet restore's before each package update.")]
+        public async Task CorrectCommandsAreExecutedForProjectFileUpdate(bool restoreBeforePackageUpdate)
+        {
+            // Arrange
+            _settingsContainer.UserSettings.RestoreBeforePackageUpdate = restoreBeforePackageUpdate;
+            var packageReferenceType = PackageReferenceType.ProjectFile;
+            var packageUpdateSet = UpdateFooFromOneVersion(packageReferenceType);
+            var sources = NuGetSources.GlobalFeed;
+
+            // Act
+            await _sut.Update(packageUpdateSet, sources);
```

Comment:
this would then take in restoreBeforePackageUpdate rather than the _settings container